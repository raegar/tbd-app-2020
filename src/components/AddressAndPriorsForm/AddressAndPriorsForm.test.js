import React from 'react';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import FormOne from './FormOne';
import { BrowserRouter as Router} from 'react-router-dom';

//cleans up 
afterEach(cleanup);

it("Renders Form One", () => {
    const { asFragment } = render(<Router><FormOne /></Router>);
    expect(asFragment()).toMatchSnapshot();
});

test('Prior Credit Conditional Rendering', () => {
    render (
        <Router>
            <FormOne />
        </Router>
    )
    //The prior credit radio button should be in the document
    expect(screen.getByTestId('prior-yes')).toBeInTheDocument()

    //The conditional textbox + caption should not be
    expect(screen.queryByText("How Many Credits?")).toBe(null)
    expect(screen.queryByTestId('prior-credit-text')).toBe(null)

    //Click the yes button
    fireEvent.click(screen.getByTestId('prior-yes'))

    //The caption + text box should now be present
    expect(screen.getByText("How Many Credits?")).toBeInTheDocument()

    //Click the no button
    fireEvent.click(screen.getByTestId('prior-no'))

    //The caption + textbox should disappear again
    expect(screen.queryByText("How many credits?")).toBe(null)
})

test('Prior Study Conditional Rendering', () => {
    render(
        <Router>
            <FormOne />
        </Router>
    )

    expect(screen.getByTestId('study-yes')).toBeInTheDocument()
    expect(screen.queryByText('When did they last study at UCP/ARU?')).toBe(null)

    fireEvent.click(screen.getByTestId('study-yes'))

    expect(screen.queryByText('When did they last study at UCP/ARU?')).toBeInTheDocument()

    fireEvent.click(screen.getByTestId('study-no'))
    expect(screen.queryByText('When did they last study at UCP/ARU?')).toBe(null)
})