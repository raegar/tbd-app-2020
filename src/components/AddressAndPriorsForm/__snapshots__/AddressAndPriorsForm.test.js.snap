// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Form One 1`] = `
<DocumentFragment>
  <div>
    <div
      class="form-container d-flex justify-content-center container"
    >
      <div
        class="form-background row"
      >
        <div
          class="col"
        >
          <h4>
            Address
          </h4>
          <div
            class="mobiletextbox"
          >
            <label
              for="address-1"
            >
              Address Line 1
            </label>
            <input
              id="address-1"
              maxlength="100"
              minlength="10"
              name="address_line_1"
              placeholder="Address Line 1"
              size="35"
              type="text"
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="address-2"
            >
              Address Line 2
            </label>
            <input
              id="address-2"
              maxlength="100"
              minlength="10"
              name="address_line_2"
              placeholder="Address Line 2"
              size="35"
              type="text"
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="town-city"
            >
              Town/City
            </label>
            <input
              id="town-city"
              maxlength="100"
              minlength="10"
              name="town_city"
              placeholder="Town/City"
              size="35"
              type="text"
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="region"
            >
              Region
            </label>
            <input
              id="region"
              maxlength="30"
              minlength="3"
              name="region"
              placeholder="Region"
              size="25"
              type="text"
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="postcode"
            >
              Postcode
            </label>
            <input
              id="postcode"
              maxlength="9"
              minlength="6"
              name="postcode"
              placeholder="Postcode"
              size="10"
              type="text"
            />
          </div>
          <br />
          <div
            class="radio-buttons"
          >
            <h4>
              Full/Part Time
            </h4>
            <p>
              Will the student study the course full-time or part-time?
            </p>
            <div
              class="radio-group"
            >
              <input
                data-testid="full"
                id="full"
                name="full-part"
                type="radio"
                value="full"
              />
              <label
                for="full"
              >
                Full Time
              </label>
            </div>
            <div
              class="radio-group"
            >
              <input
                data-testid="part"
                id="part"
                name="full-part"
                type="radio"
                value="part"
              />
              <label
                for="part"
              >
                Part Time
              </label>
            </div>
          </div>
        </div>
        <div
          class="col"
        >
          <div>
            <div
              class="radio-buttons"
            >
              <h4>
                Prior Credit
              </h4>
              <p>
                Does the student wish to enrol with prior credit?
              </p>
              <div
                class="radio-group"
              >
                <input
                  data-testid="prior-yes"
                  id="prior-yes"
                  name="prior-credit"
                  type="radio"
                  value="yes"
                />
                <label
                  for="prior-yes"
                >
                  Yes
                </label>
              </div>
              <div
                class="radio-group"
              >
                <input
                  data-testid="prior-no"
                  id="prior-no"
                  name="prior-credit"
                  type="radio"
                  value="no"
                />
                <label
                  for="prior-no"
                >
                  No
                </label>
              </div>
            </div>
          </div>
          <span />
          <div
            class="radio-buttons"
          >
            <h4>
              Year of Entry
            </h4>
            <p>
              In what year does the student want to begin their study?
            </p>
            <div
              class="radio-group"
            >
              <input
                data-testid="year-1"
                id="year-1"
                name="year-of-entry"
                type="radio"
                value="1"
              />
              <label
                for="year-1"
              >
                Year 1
              </label>
            </div>
            <div
              class="radio-group"
            >
              <input
                data-testid="year-2"
                id="year-2"
                name="year-of-entry"
                type="radio"
                value="2"
              />
              <label
                for="year-2"
              >
                Year 2
              </label>
            </div>
            <div
              class="radio-group"
            >
              <input
                data-testid="year-3"
                id="year-3"
                name="year-of-entry"
                type="radio"
                value="3"
              />
              <label
                for="year-3"
              >
                Year 3
              </label>
            </div>
          </div>
          <div
            class="radio-buttons"
          >
            <h4>
              Prior Study
            </h4>
            <p>
              Has the student studied at UCP/ARU before?
            </p>
            <div
              class="radio-group"
            >
              <input
                data-testid="study-yes"
                id="study-yes"
                name="prior-study"
                type="radio"
                value="yes"
              />
              <label
                for="study-yes"
              >
                Yes
              </label>
            </div>
            <div
              class="radio-group"
            >
              <input
                data-testid="study-no"
                id="study-no"
                name="prior-study"
                type="radio"
                value="no"
              />
              <label
                for="study-no"
              >
                No
              </label>
            </div>
          </div>
          <span />
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="row"
        id="buttonRow"
      >
        <div
          class="centered-buttons col"
        >
          <a
            href="/NewApplicationPage"
          >
            <button
              class="mediumbutton"
              data-testid="ucpButton"
              primary="True"
              type="button"
            >
              Go Back​
            </button>
          </a>
        </div>
        <div
          class="centered-buttons col"
        >
          <div
            id="confirmButton"
          >
            <a
              href="/LevelTwoQualifications"
            >
              <button
                class="mediumbutton"
                data-testid="ucpButton"
                primary="True"
                type="button"
              >
                Next​
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
